imap_processing/codice/codice_l1a.py

```diff
@@ -132,28 +141,27 @@ def create_science_dataset(self, start_time: np.datetime64) -> xr.Dataset:
                 -1, self.num_energy_steps
             )
 
-            counter_attrs = cdf_attrs.counters_attrs
-            counter_attrs["FIELDNAM"] = fieldname
-            counter_attrs["CATDESC"] = catdesc
+            cdf_attrs.variable_attributes["counters_attrs"]["FIELDNAM"] = fieldname
```

Comment:
You shouldn't be updating the dictionary like this - any logic for setting attributes should stem from the YAML files. Don't make me regret using python dicts! If you absolutely need to do it like this, create a copy of the variable attributes file and update that one instead of going back into the cdf_attrs instance.

Reply:
So instead I could define a new variable attribute for each "counter" I have here (there are 16 total in this case) that inherits the `counters_attrs` and just switches out the `CATDESC` and `FIELDNAM`. Something like this:

```yaml
counters_attrs: &counters
    <<: *default
    CATDESC: Fill in at creation
    DEPEND_0: epoch
    DEPEND_1: energy
    DISPLAY_TYPE: time_series
    FIELDNAM: Fill in at creation
    LABL_PTR_1: energy
    UNITS: counts
    VALIDMIN: 0
    VALIDMAX: 8388607  # max value for a signed 24-bit integer
    
counter1_attrs:
    <<: *counters
    CATDESC: catdesc for counter 1 of 16
    FIELDNAM: fieldname for counter 1 of 16
    
counter2_attrs:
    <<: *counters
    CATDESC: catdesc for counter 2 of 16
    FIELDNAM: fieldname for counter 2 of 16

...etc
```

Reply:
Yup, exactly. All the data should be in the YAML files, nothing added later.